libs/community/langchain_community/cache.py

```diff
@@ -1377,29 +1386,35 @@ def __init__(
             token (Optional[str]): API token for Astra DB usage.
             api_endpoint (Optional[str]): full URL to the API endpoint,
                 such as "https://<DB-ID>-us-east1.apps.astra.datastax.com".
-            astra_db_client (Optional[Any]): *alternative to token+api_endpoint*,
+            astra_db_client (Optional[AstraDB]):
+                *alternative to token+api_endpoint*,
                 you can pass an already-created 'astrapy.db.AstraDB' instance.
+            async_astra_db_client (Optional[AsyncAstraDB]):
+                *alternative to token+api_endpoint*,
+                you can pass an already-created 'astrapy.db.AsyncAstraDB' instance.
             namespace (Optional[str]): namespace (aka keyspace) where the
                 collection is created. Defaults to the database's "default namespace".
+            pre_delete_collection (bool): whether to delete and re-create the
+                collection. Defaults to False.
+            async_setup (bool): whether to create the collection asynchronously.
+                Enable only if there is a running asyncio event loop. Defaults to False.
         """
-        astra_env = AstraDBEnvironment(
+        self.astra_env = AstraDBCollectionEnvironment(
```

Comment:
FYI:  https://github.com/langchain-ai/langchain/issues/17242 -- will make it possible to set caches at request time, so we'll be more likely to see repeated instantiation of objects.

AstraDBCollectionEnvironment makes a network request on every init. OK for now, but could become expensive if user re-instantiates the cache repeatedly (which I think they should be allowed to do)

Reply:
added SetupMode param (SYNC, ASYNC, OFF)